src/core/coap/coap.hpp

```diff
@@ -664,7 +662,7 @@ class Coap : public InstanceLocator
      * @param[in]  aMessageInfo  A reference to the message info associated with @p aMessage.
      *
      */
-    virtual otError Send(Message &aMessage, const Ip6::MessageInfo &aMessageInfo);
+    virtual otError Send(ot::Message &aMessage, const Ip6::MessageInfo &aMessageInfo);
```

Comment:
How about changing this to accept `Coap::Message` instead of `ot::Message` since in both `Coap` and `CoapSecure` we typecast it to `Coap::Message`?

BTW, just curious, do you happen to know the reason for this and `Receive` being `virtual`?

Reply:
`virtual` was introduced by someone new to OpenThread. In fact, me ^_^
I submitted #3434 which removes these virtual methods.